New insert tag system

ausi opened the pull request:
See #1379

Do we usually link to source code on github for examples?
Should we add more links to such examples here? (currently only the `IfLanguageInsertTag` is linked to).

ausi committed:
Add new insert tag system

fritzmg committed:
added block insert tag description and example

fritzmg commented:
I added a little more description regarding block insert tags and also included a (beautifully) simple (and still super useful) example. wdyt?

I also changed the namespace from `InsertTags` to `InsertTag` as namespaces are usually always singular.

ausi commented on docs/dev/framework/insert-tags.md:
```diff
@@ -8,12 +8,210 @@ aliases:
+namespace App\InsertTag;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsInsertTag;
+use Contao\CoreBundle\InsertTag\InsertTagResult;
```
```suggestion
use Contao\CoreBundle\InsertTag\Exception\InvalidInsertTagException;
use Contao\CoreBundle\InsertTag\InsertTagResult;
```

ausi committed:
Add missing use statement

ausi commented on docs/dev/framework/insert-tags.md:
```diff
@@ -8,12 +8,211 @@ aliases:
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Symfony\Bundle\SecurityBundle\Security;
+
+#[AsBlockInsertTag('ifmembergroup', 'endifmembergroup')]
```
```suggestion
#[AsBlockInsertTag('ifmembergroup', endTag: 'endifmembergroup')]
```
Should we add the parameter name to make it more obvious what this refers to?

ausi commented:
> I added a little more description regarding block insert tags and also included a (beautifully) simple (and still super useful) example. wdyt?

perfect 👍

> I also changed the namespace from `InsertTags` to `InsertTag` as namespaces are usually always singular.

:+1:

ausi commented on docs/dev/framework/insert-tags.md:
````diff
@@ -8,12 +8,211 @@ aliases:
+```
+
+When implementing a block insert tag your function receives the wrapped content as a `ParsedSequence` object in addition
+to your `InsertTag`. Your function is then expected to return a `ParsedSequence` again - which could be empty in
````
```suggestion
to your `InsertTag`. The `ParsedSequence` object consists of strings and other parsed (but not yet resolved)
insert tags. Your function is then expected to return a `ParsedSequence` again - which could be empty in
```

ausi committed:
Add info about ParsedSequence

fritzmg committed:
Update docs/dev/framework/insert-tags.md

Co-authored-by: Martin Auswöger <martin@auswoeger.com>

fritzmg commented:
Thank you @ausi :)
